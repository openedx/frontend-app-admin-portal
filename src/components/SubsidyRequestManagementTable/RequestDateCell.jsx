import PropTypes from 'prop-types';

import { formatTimestamp } from '../../utils';

const RequestDateCell = ({ row }) => formatTimestamp({ timestamp: row.original.requestDate });

RequestDateCell.propTypes = {
  row: PropTypes.shape({
    original: PropTypes.shape({
      requestDate: PropTypes.string,
    }),
  }).isRequired,
};

export default RequestDateCell;
