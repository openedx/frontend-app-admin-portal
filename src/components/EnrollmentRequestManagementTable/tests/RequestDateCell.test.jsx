import React from 'react';
import renderer from 'react-test-renderer';

import RequestDateCell from '../RequestDateCell';

const defaultProps = {
  row: {
    original: {
      requestDate: '2019-12-03T21:39:24.395101Z',
    },
  },
};

describe('RequestDateCell', () => {
  test('renders as expected', () => {
    const tree = renderer
      .create(<RequestDateCell {...defaultProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
