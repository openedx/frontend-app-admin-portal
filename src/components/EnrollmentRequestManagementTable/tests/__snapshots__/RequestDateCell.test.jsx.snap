// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestDateCell renders as expected 1`] = `"December 3, 2019"`;
